fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -117,11 +117,22 @@ public LuceneIndexMaintainer(@Nonnull final IndexMaintainerState state, @Nonnull
                                          @Nonnull ScanProperties scanProperties) {
         LOG.trace("scan scanType={}", scanType);
         Verify.verify(range.getLow() != null);
-        Verify.verify(scanType == IndexScanType.BY_LUCENE);
+        Verify.verify(scanType == IndexScanType.BY_LUCENE || scanType == IndexScanType.BY_LUCENE_FULL_TEXT);
         try {
-            final QueryParser parser = new QueryParser(FULL_TEXT_KEY_FIELD, analyzer);
-            Query query = parser.parse(range.getLow().getString(0));
-            return new LuceneRecordCursor(executor, scanProperties, state, query, continuation, state.index.getRootExpression().normalizeKeyForPositions());
+            // This cannot work with nested documents the way that we currently use them. BlockJoin will be essential for this
+            // functionality in this way.
+            if (scanType == IndexScanType.BY_LUCENE_FULL_TEXT) {
+                List<String> fieldNames = listIndexFieldNames(state.index.getRootExpression());
+                MultiFieldQueryParser parser = new MultiFieldQueryParser(fieldNames.toArray(new String[fieldNames.size()]), analyzer);
+                parser.setDefaultOperator(QueryParser.Operator.OR);
+                Query query  = parser.parse(range.getLow().getString(0));
+                return new LuceneRecordCursor(executor, scanProperties, state, query, continuation, state.index.getRootExpression().normalizeKeyForPositions());
```

Comment:
This seems complicated enough to arrange to put outside the `if`. `MultiFieldQueryParser` extends `QueryParser`, right? So the conditional could just make the right kind of parser.